pkg/rbd/controllerserver.go

```diff
@@ -383,9 +393,11 @@ func (cs *ControllerServer) DeleteVolume(ctx context.Context, req *csi.DeleteVol
 	}
 	defer cs.VolumeLocks.Release(volumeID)
 
-	rbdVol := &rbdVolume{}
-	defer rbdVol.Destroy()
-	if err = genVolFromVolID(ctx, rbdVol, volumeID, cr, req.GetSecrets()); err != nil {
+	rbdVol, err := genVolFromVolID(ctx, volumeID, cr, req.GetSecrets())
+	if rbdVol != nil {
+		defer rbdVol.Destroy()
```

Comment:
do we need to Destroy if the err!=nil?

Reply:
the `if` is about `rbdVol != nil`, not `err`

Reply:
If it is needed then it is probably also needed at around line 527 below.

Reply:
I think this has been addressed already?